### ISSUE TITLE
Reuse expressions in resource

### ISSUE THREAD

**Issue by glzbcrt** [hashicorp/terraform#23253]
Hello,

In this resource I use the same expression 4 times as highlighted.
It would be nice to apply the DRY principle and define it only once and use several times.

I didn't find a way to do it inside resources where the count is used.
Is there a way? Or it should be a new feature?

resource "azurerm_virtual_machine" "appServers" {
    count                 = "${var.appNumOfInstances}"
    name                  = **"${var.appPrefix}${format("%02d", count.index + 1 + var.appLastInstanceId)}"**
    location              = "${data.azurerm_resource_group.appRG.location}"
    resource_group_name   = "${data.azurerm_resource_group.appRG.name}"
    network_interface_ids = ["${element(azurerm_network_interface.appInterfaces.*.id, count.index)}"]
    vm_size               = "${var.appSizes["**${var.appPrefix}${format("%02d", count.index + 1 + var.appLastInstanceId)}**"]}"

    delete_os_disk_on_termination = true

    storage_image_reference {
        publisher = "${var.appOsImage["publisher"]}"
        offer     = "${var.appOsImage["offer"]}"
        sku       = "${var.appOsImage["sku"]}"
        version   = "${var.appOsImage["version"]}"
    }

    storage_os_disk {
        name              = **"${lower(var.appPrefix)}${format("%02d", count.index + 1 + var.appLastInstanceId)}**-os"
        caching           = "ReadWrite"
        create_option     = "FromImage"
        managed_disk_type = "StandardSSD_LRS"
        disk_size_gb      = "32"
    }

    os_profile {
        computer_name  = **"${var.appPrefix}${format("%02d", count.index + 1 + var.appLastInstanceId)}"**
        admin_username = "${var.appCredentials["username"]}"
        admin_password = "${var.appCredentials["password"]}"
    }
}

Thanks,

**Comment by hashibot** [comment]
Hello! :wave: 

This issue doesn't seem to be following one of the standard issue templates available from [the "New Issue" page](https://github.com/hashicorp/terraform/issues/new/choose). The information requested in those templates is important for my human friends to fully understand what's going on.

Please feel free to open a new issue using either the Bug Report or Feature Request template, as appropriate, and then the Terraform maintainers will review it. Thanks!

Please note that we use GitHub issues for tracking bugs and enhancements rather than for questions. While we may be able to help with certain simple problems here it's better to use [the community forum](https://discuss.hashicorp.com/c/terraform-core), where there are more people ready to help. The GitHub issues here are generally monitored only by our few core maintainers.
